lib/createBundle.js

```diff
@@ -15,23 +15,38 @@ const fetch = require('./fetch');
  * @param  {String}  targetPath The destination path where the bundle will be written
  * @param  {Boolean} dev        Whether to build the bundle in development mode
  * @param  {Boolean} minify     Whether to minify the bundle
+ * @param  {Boolean} sourceMap  Whether to generate a source map (with a `.map` suffix on `targetPath`)
  * @return {Promise}            Resolved when the bundle is written to the target path
  */
 module.exports = function createBundle(server, options) {
-  const bundleUrl = url.format({
+  let promise = download(buildUrl(server, options, 'bundle'), options.targetPath);
+  if (options.sourceMap) {
+    promise = promise.then(() => {
+      download(buildUrl(server, options, 'map'), `${options.targetPath}.map`);
+    });
+  }
+
+  return promise;
```

Comment:
Can you instead construct two promises (for bundle & map) and return `Promise.all()`? Something along the lines of
```js
const bundlePromise =
  fetchBundle(server, options)
    .then(content => writeFile(options.targetPath, content));

const mapPromise =
  !options.sourceMap ?
    Promise.resolve() :
    fetchMap(server, options)
      .then(content => writeFile(`${options.targetPath}.map`, content));

return Promise.all([bundlePromise, mapPromise]);
```